durable-storage/src/persistence_layer.rs

```diff
@@ -0,0 +1,170 @@
+// SPDX-FileCopyrightText: 2025 Trilitech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+//! Implementation of the RISC-V PVM durable storage's persistence layer.
+//!
+//! A persistence layer is tied to a repository on disk, identified by a directory.
+//! Within that directory, the persistence layer needs to be able to perform basic KV operations:
+//! - get
+//! - set
+//! - delete
+//!
+//! As well as repository-level operations:
+//! - new
+//! - clone
+//! - commit (returning the commit hash)
+//! - checkout a specific commit
+//!
+//! The folder structure of the [`DirectoryManager`] is:
+//! ```
+//! <repo_path>:
+//!    temporary/
+//!        db_<random>/checkpoint/
+//!            <rocksdb internals>
+//! ```
+
+use std::mem::ManuallyDrop;
+
+use crate::repo::DirectoryManager;
+use crate::repo::DirectoryManagerError;
+
+/// Type alias for a 32-byte hash used for identifying key-value blobs & commits.
+type Hash = [u8; 32];
+
+/// Errors encountered when interacting with the persistence layer.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("RocksDB error: {0}")]
+    RocksDB(#[from] rocksdb::Error),
+
+    #[error("Directory manager error: {0}")]
+    DirectoryManager(#[from] DirectoryManagerError),
+}
+
+/// Create a [`rocksdb::Options`] instance for creating a new database.
+fn options_expect_newly_created() -> rocksdb::Options {
```

Comment:
Will this be used anywhere except to initialise a `PersistenceLayer`? If not, I think it should either be folded in `PersistenceLayer::new` or be part of its `impl`.

Reply:
Folded it in the `new` method.